website/blog/2022-12-22-block-patterns.md

```diff
@@ -0,0 +1,132 @@
+```shell
+wp boilerplate create service-example --folder=BlockPatterns --file_name=ManagePatternCategories
+```
+
+Now that we have our service class, we'll add methods for removing core patterns and pattern categories. First, the method for removing core categories.
+```php
+/**
+ * Unregisters core block pattern categories
+ *
+ * @return void
+ */
+public function removeCoreCategories()
+{
+	\unregister_block_pattern_category('buttons');
+	\unregister_block_pattern_category('columns');
+	\unregister_block_pattern_category('gallery');
+	\unregister_block_pattern_category('header');
+	\unregister_block_pattern_category('text');
+}
+```
+The second method will remove all core patterns
+
+```php
+/**
+ * Removes support for core block patterns
+ *
+ * @return void
+ */
+public function removeBlockPatternsCore()
+{
+	\remove_theme_support('core-block-patterns');
+}
+```
+
+For these methods to work, we need to add them to the `register()` method of our service class. Add the following actions:
+```php
+\add_action('init', [$this, 'removeCoreCategories'], 40);
+\add_action('after_setup_theme', [$this, 'removeBlockPatternsCore'], 20);
+```
+
+If you try searching for patterns in your editor, you'll notice they are no longer available. Now we can move on to adding our block pattern categories. The best approach is to define the pattern category names as constants inside your class. We will add two categories, **Templates** and **Sections**. We can add the following code inside our class:
+```php
+class ManagePatternCategories implements ServiceInterface
+{
+	public const TEMPLATES_CATEGORY = 'templates-category';
+	public const SECTIONS_CATEGORY = 'sections-category';
+
+	// Removed parts of code for better readability.
+
+	/**
+	 * Removes support for core block patterns
+	 *
+	 * @return void
+	 */
+	public function addCategories()
+	{
+		\register_block_pattern_category(self::TEMPLATES_CATEGORY, [
+			'label' =>  \esc_html__('Templates', 'es-blog'),
+		]);
+
+		\register_block_pattern_category(self::SECTIONS_CATEGORY, [
+			'label' => \esc_html__('Sections', 'es-blog'),
+		]);
+	}
+}
+```
+
+Finally, return to your Intro Pattern class and update the `getCategories()` method:
+```php
+protected function getCategories(): array
+{
+	return [
+		ManagePatternCategories::SECTIONS_CATEGORY
+	];
+}
+```
+
+In editor, in the Pattern tab, you should now see your new pattern category and the Intro Pattern.
+
+![New pattern category and block pattern](/img/blog/block-pattern-example.png)
+
+## Things to keep in mind
+Since Block Patterns just add content which you can then modify at will, it becomes an issue if you've already used this pattern on multiple places and then notice you missed something. Fixing the pattern in code will apply the changes on all new patterns you add, but you will have to fix this manually in the editor everywhere where you've already used that pattern.
```

Comment:
```suggestion
Block patterns add layout and content which you can then modify at will per instance. These instances are not synchronised with the code-base so it becomes an issue if you've already used a pattern on multiple places on the site and then notice you've missed something in the pattern configuration phase. Fixing the pattern in the code-base will apply the changes on all newly added instances, but you will have to fix the misconfigured instances manually.
```

My attempt at rewording this part to be more specific :)

Reply:
An excellent attempt I must say 😄 💪